### PR TITLE
python311Packages.jupyterlab-git: 0.42.0 -> 0.50.0rc0

### PR DESCRIPTION
**Pull request by natsukium**
## Description of changes

Diff: https://github.com/jupyterlab/jupyterlab-git/compare/v0.42.0...v0.50.0rc0

Changelog: https://github.com/jupyterlab/jupyterlab-git/blob/v0.50.0rc0/CHANGELOG.md

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by natsukium** [comment]
wait for v0.50.0
https://github.com/jupyterlab/jupyterlab-git/pull/1236

File: pkgs/development/python-modules/jupyterlab-git/default.nix [hunk 1]
```diff
@@ -72,5 +78,7 @@ buildPythonPackage rec {
     homepage = "https://github.com/jupyterlab/jupyterlab-git";
     license = with licenses; [ bsd3 ];
     maintainers = with maintainers; [ chiroptical ];
+    # https://github.com/jupyterlab/jupyterlab-git/issues/1245
```
Comment (kirillrdy) [inline pkgs/development/python-modules/jupyterlab-git/default.nix:81]: [0.50.0a2](https://github.com/jupyterlab/jupyterlab-git/releases/tag/v0.50.0a2) supports lupyerlab-4

Comment (natsukium) [reply pkgs/development/python-modules/jupyterlab-git/default.nix:81]: I will wait for the official release 0.50.0, but may adopt it in the ZHF.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 262742` at 8dcd50a2 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python310Packages.jupyterlab-git</li>
    <li>python311Packages.jupyterlab-git</li>
  </ul>
</details>

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 262742` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.jupyterlab-git</li>
    <li>python310Packages.jupyterlab-git.dist</li>
    <li>python311Packages.jupyterlab-git</li>
    <li>python311Packages.jupyterlab-git.dist</li>
  </ul>
</details>

**Comment by natsukium** [comment]
0.50.0 was just released.😇
https://github.com/jupyterlab/jupyterlab-git/releases/tag/v0.50.0

**Pull request merged**